ts/Core/Series/DataLabel.ts

```diff
@@ -608,14 +608,37 @@ namespace DataLabel {
                                 point.contrastColor = renderer.getContrast(
                                     (point.color || series.color) as any
                                 );
+
+                                const background = chart.chartBackground,
+                                    plotBackground = chart.plotBackground;
+
+                                let contrastColor: string =
+                                    Palette.neutralColor100;
+
+                                if (
+                                    plotBackground &&
+                                    plotBackground.attr('fill') !== 'none'
+                                ) {
+                                    contrastColor = renderer.getContrast(
+                                        plotBackground.attr('fill') as any
+                                    );
+                                } else if (
+                                    background &&
+                                    background.attr('fill') !== 'none'
+                                ) {
+                                    contrastColor = renderer.getContrast(
+                                        background.attr('fill') as any
+                                    );
+                                }
+
```

Comment:
I think it would be better to read the colors directly from the options: 

```ts
                                const contrastColor = renderer.getContrast(pick(
                                    chart.options.chart.plotBackgroundColor,
                                    chart.options.chart.backgroundColor,
                                    Palette.neutralColor10
                                ));
```

I get some type issues with this, but those are the same as you currently have with `.attr('fill')`. The problem is that the colors can be other types than strings. Maybe we should just type-check the options because gradients are hard to handle anyway.

Also, the chart and plot background is the same for all points, so we should move this calculation out of the `forEach` loop.